### ISSUE TITLE
about the raw-feature-extractor

### ISSUE THREAD

**Issue by zzjss12** [Yunlongs/Genius#1]
Do i only need to run batch_run. py to extract the control flow chart?  Will it call other functions? After I changed the path in (batch_run. Py), I tried to run it, but I found that it seemed to fall into an endless loop.

At the same time, I also found that the paths in other files(Under raw_feature_extractor) seem to be different from my local ones. Do I need to modify them,too?

I would appreciate it if you could help me！

**Comment by zzjss12** [comment]
![image](https://user-images.githubusercontent.com/66284098/165304658-fd315d50-c4ef-42f5-986b-a87252cad1b9.png)
are these functions not found because of the version of IDA,or other reasons?

**Comment by Yunlongs** [comment]
>

**Comment by Yunlongs** [comment]
> Do i only need to run batch_run. py to extract the control flow chart? Will it call other functions? After I changed the path in (batch_run. Py), I tried to run it, but I found that it seemed to fall into an endless loop.

Yes, run only `batch_run.py` is enough. It will call other modules in other python files. You should make sure the path is correct and IDA version is v7.0. 

If you want to have a deeper understanding about these scripts, I recommend you read the book :《IDA Pro权威指南第二版 》

**Comment by Yunlongs** [comment]
> At the same time, I also found that the paths in other files(Under raw_feature_extractor) seem to be different from my local ones. Do I need to modify them,too?

Ideally, you needn't to change the paths in other files. Because in these scripts, the paths is used in `__main__` function which used to run a this script for debugging. If they are imported by other files, the codes in `__main__` will not run.

**Comment by Yunlongs** [comment]
> ![image](https://user-images.githubusercontent.com/66284098/165304658-fd315d50-c4ef-42f5-986b-a87252cad1b9.png) are these functions not found because of the version of IDA,or other reasons?

Which functions? `GetMnem` and `NextHead` are IDA APIs, if they are not found please check whether your IDA version is V7.0.

**Comment by zzjss12** [comment]
Thank you for your help. After I switched IDA to version 7.0, those bugs did disappear.
But when I execute batch_ Run file, the program has been running without any results. Do you know why?

In your request: Networkx > 2.3 in PY3, but (batch_run. Py) should be executed with (python2). Isn't that contradictory?

**Comment by Excelsior-007** [comment]
I also met the same problem, after I installed IDA 7.0, batch_run.py can be executed without any error, but I got nothing from the program. I don't know where the problem is. Dose *.exe file can be used to extract CFG?
Also, I want to know what "start = 0" means.
Thank you very much.

**Comment by Excelsior-007** [comment]
> Thank you for your help. After I switched IDA to version 7.0, those bugs did disappear. But when I execute batch_ Run file, the program has been running without any results. Do you know why?
> 
> In your request: Networkx > 2.3 in PY3, but (batch_run. Py) should be executed with (python2). Isn't that contradictory?

Have you solved this problem?,I executed the batch_run.py but also get nothing in the extracted-acfg folder.

**Comment by Hululuhh** [comment]
> > Thank you for your help. After I switched IDA to version 7.0, those bugs did disappear. But when I execute batch_ Run file, the program has been running without any results. Do you know why?
> > In your request: Networkx > 2.3 in PY3, but (batch_run. Py) should be executed with (python2). Isn't that contradictory?
> 
> Have you solved this problem?,I executed the batch_run.py but also get nothing in the extracted-acfg folder.

Is there any solution?I have the same problem.

**Comment by Hululuhh** [comment]
Will the author read the comments? We need your help. Thank you very much.

**Comment by Excelsior-007** [comment]
> > > Thank you for your help. After I switched IDA to version 7.0, those bugs did disappear. But when I execute batch_ Run file, the program has been running without any results. Do you know why?
> > > In your request: Networkx > 2.3 in PY3, but (batch_run. Py) should be executed with (python2). Isn't that contradictory?
> > 
> > 
> > Have you solved this problem?,I executed the batch_run.py but also get nothing in the extracted-acfg folder.
> 
> Is there any solution?I have the same problem.

The auther has not responded,and I have already given up this project.

**Comment by Hululuhh** [comment]
I need ACFG information. I'll look for a solution tomorrow. I'm falling apart. ---- Replied Message ----FromJiahao ***@***.***>Date09/16/2023 14:52 ***@***.***> ***@***.***>***@***.***>SubjectRe: [Yunlongs/Genius] about the raw-feature-extractor (Issue #1)



Thank you for your help. After I switched IDA to version 7.0, those bugs did disappear. But when I execute batch_ Run file, the program has been running without any results. Do you know why?
In your request: Networkx > 2.3 in PY3, but (batch_run. Py) should be executed with (python2). Isn't that contradictory?

Have you solved this problem?,I executed the batch_run.py but also get nothing in the extracted-acfg folder.

Is there any solution?I have the same problem.

The auther has not responded,and I have already given up this project.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/Yunlongs/Genius/issues/1#issuecomment-1722156538",
"url": "https://github.com/Yunlongs/Genius/issues/1#issuecomment-1722156538",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
